homeassistant/components/weatherkit/__init__.py

```diff
@@ -0,0 +1,49 @@
+"""Integration for Apple's WeatherKit API."""
+from __future__ import annotations
+
+from apple_weatherkit.client import WeatherKitApiClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import CONF_KEY_ID, CONF_KEY_PEM, CONF_SERVICE_ID, CONF_TEAM_ID, DOMAIN
+from .coordinator import WeatherKitDataUpdateCoordinator
+
+PLATFORMS: list[Platform] = [Platform.WEATHER]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up this integration using UI."""
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = coordinator = WeatherKitDataUpdateCoordinator(
```

Comment:
Please first create the coordinator, then execute the first refresh, only then you may add it to the hass object.